server/pbench/bin/pbench-report-status

```diff
@@ -86,36 +75,52 @@ rm -rf $tmp" EXIT QUIT INT
 mkdir $tmp
 
 server=$(getconf.py --config $IDXCONFIG server Server)
-if [[ -z "$server" ]] ;then
-    echo "$prog: no server defined in [Server] section of $IDXCONFIG" >> ${tmp}/log
-    exit 2
-fi
+index_prefix=$(getconf.py --config $IDXCONFIG index_prefix Settings)
+if [ -z "$server" -o -z "$index_prefix" ] ;then
+    # We don't have an Elasticsearch configuration, use syslog.
+    if [ -z "$server" -o -z "$index_prefix" ] ;then
+        echo "$prog: no server defined in [Server] section of $IDXCONFIG" >> ${tmp}/log
+    fi
+    if [ -z "$server" -o -z "$index_prefix" ] ;then
+        echo "$prog: no index_prefix defined in [Settings] section of $IDXCONFIG" >> ${tmp}/log
+    fi
 
-# Snip off the trailing "-<TZ>" - assumes that <TZ> does not contain a "-"
-datestamp="${timestamp%-*}"
-# Snip off the time part of the timestamp to get YYYY-MM
-datestamp=${datestamp%-*}
+    # Prepend the proper sequence to allow rsyslog to recognize JSON and process it
+    printf "@cee:" > ${tmp}/payload
+    # Generate the payload without pretty JSON format so it fits in a single line.
+    $dir/pbench-gen-json-payload ${timestamp%-*} $name $doctype $file "false" >> ${tmp}/payload
 
-index=$(getconf.py --config $IDXCONFIG index_prefix Settings)."$name"."$datestamp"
-if [[ -z "$index" ]] ;then
-    echo "$prog: no index_prefix defined in [Settings] section of $IDXCONFIG" >> ${tmp}/log
-    exit 3
-fi
+    size=$(stat --printf="%s" "${tmp}/payload")
+    if [ $size -gt 4096 ]; then
+        xz -9 -c ${tmp}/payload > $LOGSDIR/$name/${prog}-payload.${timestamp%-*}.xz
+    else
+        cat ${tmp}/payload | logger --size 4096 --tag $name --priority daemon.info
+    fi
+else
+    # We have an Elasticsearch configuration.
+
+    # Snip off the trailing "-<TZ>" - assumes that <TZ> does not contain a "-"
+    datestamp="${timestamp%-*}"
+    # Snip off the time part of the timestamp to get YYYY-MM
```

Comment:
This threw me: maybe add a comment documenting the format of the timestamp? Also change the comment to say "... the day and time part..." perhaps.